heat/core/tests/test_suites/basic_test.py

```diff
@@ -0,0 +1,173 @@
+        distributed_result: bool, optional
+            Specify whether the result of the heat function is distributed across all nodes or all nodes have the full
+            result. Default is True.
+        heat_args: dictionary, optional
+            The keyword arguments that will be passed to the heat function. Array and split function don't need to be
+            specified. Default is {}.
+        numpy_args:
+            The keyword arguments that will be passed to the numpy function. Array doesn't need to be specified.
+            Default is {}.
+
+        Raises
+        ------
+        AssertionError if the functions to not perform equally.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> import heat as ht
+        >>> a = np.arange(10)
+        >>> self.assert_func_equal(a, ht.exp, np.exp)
+
+        >>> self.assert_func_equal(a, ht.exp, np.log)
+        AssertionError: [...]
+        >>> self.assert_func_equal((1, 3, 5), ht.any, np.any, distributed_result=False)
+
+        >>> heat_args = {'sorted': True, 'axis': 0}
+        >>> numpy_args = {'axis': 0}
+        >>> self.assert_func_equal([5, 5, 5, 5], ht.unique, np.unique, heat_arg=heat_args, numpy_args=numpy_args)
+        """
+        self.assertTrue(callable(heat_func))
+        self.assertTrue(callable(numpy_func))
+        if isinstance(tensor, tuple) or isinstance(tensor, list):
+            tensor = self._create_random_array(tensor)
+
+        if heat_args is None:
+            heat_args = {}
+        if numpy_args is None:
+            numpy_args = {}
+
+        if isinstance(tensor, np.ndarray):
+            torch_tensor = torch.from_numpy(tensor.copy())
+            np_array = tensor
+        elif isinstance(tensor, torch.Tensor):
+            torch_tensor = tensor
+            np_array = tensor.numpy().copy()
+        else:
+            raise TypeError('The input tensors type must be one of [tuple, list, ' +
+                            'numpy.ndarray, torch.tensor] but is {}'.format(type(tensor)))
+
+        np_res = numpy_func(np_array, **numpy_args)
+        if not isinstance(np_res, np.ndarray):
+            np_res = np.array([np_res])
+
+        dtype = types.canonical_heat_type(torch_tensor.dtype)
+        for i in range(len(tensor.shape)):
+            ht_array = factories.array(torch_tensor, split=i, dtype=dtype, device=self.device, comm=self.comm)
+            ht_res = heat_func(ht_array, **heat_args)
+            if distributed_result:
+                self.assert_array_equal(ht_res, np_res)
+            else:
+                self.assertTrue(np.array_equal(ht_res._DNDarray__array.numpy(), np_res))
+
+    def _create_random_array(self, shape):
+        # Ensure all processes have the same initial array
+        if self.get_rank() == 0:
+            array = np.random.randn(*shape)
+        else:
+            array = np.empty(shape)
+        self.comm.Bcast(array, root=0)
```

Comment:
couldnt this be avoided by setting a manual seed? then every array will be the same on all nodes anyway